src/sentry/incidents/models.py

```diff
@@ -24,13 +29,49 @@ class Meta:
         unique_together = (('project', 'incident'), )
 
 
+class IncidentGroup(Model):
+    __core__ = False
+
+    group = FlexibleForeignKey('sentry.Group', db_index=False, db_constraint=False)
+    incident = FlexibleForeignKey('sentry.Incident')
+
+    class Meta:
+        app_label = 'sentry'
+        db_table = 'sentry_incidentgroup'
+        unique_together = (('group', 'incident'), )
+
+
 class IncidentManager(BaseManager):
     def fetch_for_organization(self, organization, projects):
         return self.filter(
             organization=organization,
             projects__in=projects,
         )
 
+    @TimedRetryPolicy.wrap(timeout=5, exceptions=(IntegrityError, ))
+    def create(self, organization, **kwargs):
+        """
+        Creates an Incident. Fetches the maximum identifier value for the org
+        and increments it by one. If two incidents are created for the
+        Organization at the same time then an integrity error will be thrown,
+        and we'll retry again several times. I prefer to lock optimistically
+        here since if we're creating multiple Incidents a second for an
+        Organization then we're likely failing at making Incidents useful.
+        """
+        with transaction.atomic():
+            result = self.filter(organization=organization).aggregate(models.Max('identifier'))
```

Comment:
What does this incrementing identifier do that the serial id column doesn't do?

Reply:
This is actually from https://github.com/getsentry/sentry/pull/13028/files, I must have just messed something up with the dependent branch. But the difference is that a serial column is a global increment, and this increments within a given organization.